controllers/platformnetwork_controller.go

```diff
@@ -1248,6 +1248,23 @@ func (r *PlatformNetworkReconciler) Reconcile(ctx context.Context, request ctrl.
 		return reconcile.Result{}, err
 	}
 
+	// Restore the PlatformNetwork status
+	if r.checkRestoreInProgress(instance) {
+		r.ReconcilerEventLogger.NormalEvent(instance, common.ResourceUpdated, "Restoring '%s' PlatformNetwork resource status without doing actual reconciliation", instance.Name)
+		if err := r.RestorePlatformNetworkStatus(instance); err != nil {
+			return reconcile.Result{}, err
+		}
+		if err := r.ClearAnnotation(instance); err != nil {
+			return reconcile.Result{}, err
+		}
+		return ctrl.Result{}, nil
+	}
+
+	if instance.Status.ObservedGeneration == instance.ObjectMeta.Generation &&
+		instance.Status.Reconciled == true {
```

Comment:
Just keep it as `instance.Status.Reconciled` instead of `instance.Stauts.Reconciled == true`.
This will also fix golang-lint issues.

Reply:
Changes are done